Client/Frontend/Browser/TabTrayController.swift

```diff
@@ -497,50 +251,69 @@ class TabTrayController: UIViewController {
         return privateMode && tabManager.privateTabs.count == 0
     }
 
-    func changePrivacyMode(_ isPrivate: Bool) {
-        if isPrivate != privateMode {
-            guard let _ = collectionView else {
-                privateMode = isPrivate
-                return
-            }
-            didTogglePrivateMode()
-        }
+    @objc func openTab() {
+        openNewTab()
     }
 
-    func openNewTab() {
+    func openNewTab(_ request: URLRequest? = nil) {
+        self.tabManager.addTabAndSelect(request, isPrivate: tabDisplayManager.isPrivate)
+        self.tabDisplayManager.performTabUpdates {
+            self.emptyPrivateTabsView.isHidden = !self.privateTabsAreEmpty()
+            self.dismissTabTray()
+        }
         LeanPlumClient.shared.track(event: .openedNewTab, withParameters: ["Source": "Tab Tray"])
-        openNewTab(nil)
     }
 
-    fileprivate func openNewTab(_ request: URLRequest?) {
-        toolbar.isUserInteractionEnabled = false
-
-        // We're only doing one update here, but using a batch update lets us delay selecting the tab
-        // until after its insert animation finishes.
-        var tab: Tab?
-        self.collectionView.performBatchUpdates({
-            tab = self.tabManager.addTab(request, isPrivate: self.privateMode)
-        }, completion: { finished in
-            // The addTab delegate method will pop to the BVC no need to do anything here.
-            self.toolbar.isUserInteractionEnabled = true
-            if finished, request == nil, NewTabAccessors.getNewTabPage(self.profile.prefs) == .blankPage,
-                let appDelegate = UIApplication.shared.delegate as? AppDelegate,
-                let bvc = appDelegate.browserViewController {
-                DispatchQueue.main.asyncAfter(deadline: .now() + 0.6) {
-                    bvc.urlBar.tabLocationViewDidTapLocation(bvc.urlBar.locationView)
-                }
-            }
+}
 
-            if let tab = tab {
-                self.delegate?.tabTrayDidAddTab(self, tab: tab)
-            }
-        })
+extension TabTrayController: TabDisplayer {
+
+    func focusSelectedTab() {
+        self.focusTab()
+    }
+
+    func cellFactory(for cell: UICollectionViewCell, using tab: Tab) -> UICollectionViewCell {
+        guard let tabCell = cell as? TabCell else { return UICollectionViewCell() }
+        tabCell.animator.delegate = self
+        tabCell.delegate = self
+        let selected = tab == tabManager.selectedTab
+        tabCell.configureWith(tab: tab, is: selected)
+        return tabCell
+    }
+}
+
+extension TabTrayController {
+
+    @objc func didTapLearnMore() {
+        let appVersion = Bundle.main.object(forInfoDictionaryKey: "CFBundleShortVersionString") as? String
+        if let langID = Locale.preferredLanguages.first {
+            let learnMoreRequest = URLRequest(url: "https://support.mozilla.org/1/mobile/\(appVersion ?? "0.0")/iOS/\(langID)/private-browsing-ios".asURL!)
```

Comment:
I know it was like this before, but I wonder if we should extract this out somewhere like a string. Maybe not though.

Reply:
If this was a simple static string I would. But we use the appVersion and locale to build this URL. I think it makes sense to just leave it here.